torch/_higher_order_ops/cond.py

```diff
@@ -28,6 +30,30 @@
 )
 
 
+@contextmanager
+def _set_compilation_env():
+    _old_is_tracing = torch.fx._symbolic_trace._is_fx_tracing_flag
+    _old_capture_dyn_ops = torch._dynamo.config.capture_dynamic_output_shape_ops
+    _old_capture_scalar = torch._dynamo.config.capture_scalar_outputs
+    _old_size_limit = torch._dynamo.config.cache_size_limit
+
+    try:
+        # We need to turn off the is_fx_tracing_flag. Remove this flag check from dyanmo
+        # once we are confident fx tracing works with dynamo.
+        torch.fx._symbolic_trace._is_fx_tracing_flag = False
+        torch._dynamo.config.capture_dynamic_output_shape_ops = True
+        torch._dynamo.config.capture_scalar_outputs = True
+        # We need to enlarge the dynamo cache size limit because
+        # cond_wrapper's frame is expected to be hitted with different inputs frequently
+        torch._dynamo.config.cache_size_limit = 256
```

Comment:
Is this just for testing? Or is there something else that we need to worry about?

Reply:
I think it's only for testing. I don't expect users to call that many amount of times. One thing in my mind is that if the cahce_size_limit hits, dynamo will throw an unimplemented, which will be turned into an error in fullgraph mode. So we're probably safe.